From: Carlos Maddela <maddela@labyrinth.net.au>
Date: Mon, 16 Mar 2015 04:55:39 +1100
Subject: Fixed support for SPfilePatternEx and improved documentation

Description: Fixed support for SPfilePatternEx and improved documentation
Author: Carlos Maddela <maddela@labyrinth.net.au>
Origin: vendor
Forwarded: no
Last-Update: 2015-03-16
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
---
 conf/acng.conf | 36 ++++++++++++++++++++----------------
 source/acfg.cc |  2 +-
 2 files changed, 21 insertions(+), 17 deletions(-)

diff --git a/conf/acng.conf b/conf/acng.conf
index 02943bd..66a0421 100644
--- a/conf/acng.conf
+++ b/conf/acng.conf
@@ -170,27 +170,31 @@ ExTreshold: 4
 # Pigeonholing files (like static vs. volatile contents) is done by (extended)
 # regular expressions.
 #
-# There are following patterns for different classes and purposes:
-# - static data that doesn't change silently ont he server (PFilePattern)
-# - volatile data that can be changed like every hour (VFilePattern)
-# - special static data that shared some file names with volatile data,
-#   and in doubt should be identified as static (SPfilePattern)
-# - a "whitelist pattern" with hints for the regular expiration job telling
-#   to keep the files even if they are not referenced by others, like crypto
-#   signatures with which clients begin their downloads (WfilePattern)
-#
-# There are two versions, the pattern variables mentioned above should not be
-# set without good reasons because they override the built-in defaults (that
-# might impact updates to future versions of apt-cacher-ng). There are also
-# versions of those patterns with Ex patterns which should be modified by local
-# administrator. They are evaluated in addition to regular patterns at runtime.
+# The following patterns are available for the purposes detailed, where
+# the latter takes precedence over the former:
+# - «PFilePattern» for static data that doesn't change silently on the server.
+# - «VFilePattern» for volatile data that may change like every hour. Files
+#   that match both PFilePattern and VfilePattern will be treated as volatile.
+# - Static data with file names that match VFilePattern may be overriden being
+#   treated as volatile by making it match the special static data pattern,
+#   «SPfilePattern».
+# - «WfilePattern» specifies a "whitelist pattern" for the regular expiration
+#   job, telling it to keep the files even if they are not referenced by
+#   others, like crypto signatures with which clients begin their downloads.
+#
+# There are two versions. The pattern variables mentioned above should not be
+# set without good reason, because they would override the built-in defaults
+# (that might impact updates to future versions of apt-cacher-ng). There are
+# also versions of those patterns ending with Ex, which may be modified by the
+# local administrator. They are evaluated in addition to the regular patterns
+# at runtime.
 #
 # To see examples of the expected syntax, run: apt-cacher-ng -p debug=1
 #
-# VfilePatternEx:
 # PfilePatternEx:
-# WfilePatternEx:
+# VfilePatternEx:
 # SPfilePatternEx:
+# WfilePatternEx:
 #
 ###############################################################################
 
diff --git a/source/acfg.cc b/source/acfg.cc
index 70deff5..6139a7c 100644
--- a/source/acfg.cc
+++ b/source/acfg.cc
@@ -102,7 +102,7 @@ MapNameToString n2sTbl[] = {
 		,{  "VfilePatternEx",          &vfilepatEx}
 		,{  "PfilePatternEx",          &pfilepatEx}
 		,{  "WfilePatternEx",          &wfilepatEx}
-		,{  "SPfilePattern",           &spfilepatEx}
+		,{  "SPfilePatternEx",         &spfilepatEx}
 //		,{  "AdminAuth",               &adminauth}
 		,{  "BindAddress",             &bindaddr}
 		,{  "UserAgent",               &agentname}
